test/config/fixtures/console-batch/basic.test.ts

```diff
@@ -0,0 +1,37 @@
+import { afterAll, afterEach, beforeAll, beforeEach, describe, test } from 'vitest'
+
+beforeAll(() => {
+  console.log('[beforeAll 1]')
+})
+beforeAll(() => {
+  console.log('[beforeAll 2]')
+})
+
+afterAll(() => {
+  console.log('[afterAll 1]')
+})
+afterAll(() => {
+  console.log('[afterAll 2]')
+})
+
+beforeEach(() => {
+  console.log('[beforeEach 1]')
+})
+beforeEach(() => {
+  console.log('[beforeEach 2]')
+})
+
+afterEach(() => {
+  console.log('[afterEach 1]')
+})
+afterEach(() => {
+  console.log('[afterEach 2]')
+})
+
+test('test', async () => {
+  console.log('[test 1]')
+  console.log('[test 2]')
+  await Promise.resolve()
+  console.log('[test 3]')
+  console.log('[test 4]')
+})
```

Comment:
It turns out separation between `before/after/test` logs are already "fixed" by https://github.com/vitest-dev/vitest/pull/6944, which added `setTimeout` wrapper. What this PR changes is only to split these 4 logs into 2-2. 

_before_

```
stdout | basic.test.ts > test
[test 1]
[test 2]
[test 3]
[test 4]
```

_after_

```
stdout | basic.test.ts > test
[test 1]
[test 2]

stdout | basic.test.ts > test
[test 3]
[test 4]
```